### PR TITLE
Switching to student view produces an error alert message

### PR DESCRIPTION
**Pull request by idevainstr**
refs: MBL-15676
affects: Student
release note: Fixed the error message while switching to Student view.
test plan: see ticket

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/100121504/p/81450a76305ea9e76708a61d1d369ac4">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/100121515/p/8fcdd0ab7c2c307da01f966d8b5481cb">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/100121495/p/daa76d4135fc22f66adc8a93ecc96345">
    
  </a>
</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: XCTest failed: CoreTests/AttachmentCopyServiceTests/testFailureStateOnInvalidAppBundle</summary>


    Assertion Failure at AttachmentCopyServiceTests.swift:62: Asynchronous wait failed: Exceeded timeout of 0.5 seconds, with unfulfilled expectations: "State updated".
    Assertion Failure at AttachmentCopyServiceTests.swift:66: failed - Invalid state Optional(Core.AttachmentCopyService.State.loading) received.
</details>
<details><summary>:x: XCTest failed: StudentUnitTests/UrlSubmissionPresenterTests/testSubmitSuccess</summary>


    Assertion Failure at UrlSubmissionPresenterTests.swift:70: Asynchronous wait failed: Exceeded timeout of 10 seconds, with unfulfilled expectations: "dismissed".
</details>


[MBL-15676](https://instructure.atlassian.net/browse/MBL-15676)

#### Affected Apps: Student

#### Release Note: 
Fixed the error message while switching to Student view.
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Notifications/NotificationManager.swift [hunk 1]
```diff
@@ -85,6 +85,9 @@ extension NotificationManager {
     }
 
     public func subscribeToPushChannel(token: Data? = nil, session: LoginSession? = AppEnvironment.shared.currentSession) {
+        guard AppEnvironment.shared.currentSession?.masquerader?.absoluteString.contains("fake-students") == nil else {
```
Comment (vargaat) [inline Core/Core/Notifications/NotificationManager.swift:88]: There is an `isFakeStudent` method on `LoginSession`, can't we use that?

Comment (idevainstr) [reply Core/Core/Notifications/NotificationManager.swift]: done

**Review by vargaat — APPROVED** [review]
QA+1

**Pull request merged**